polyfill/lib/calendar.mjs

```diff
@@ -171,15 +171,39 @@ DefineIntrinsic('Temporal.Calendar.prototype.toString', Calendar.prototype.toStr
 
 impl['iso8601'] = {
   dateFromFields(fields, overflow) {
-    const { year, month, day } = ES.ToRecord(fields, [['day'], ['month'], ['year']]);
```

Comment:
I think a similar change needs to be made below in the `gregory` and `japanese` calendars.

Reply:
The reason I made this change was to ensure that `ISO{Date,...}FromFields` preserve `[get, valueOf, get, valueOf, ...]` order in their accesses. Custom calendars don't need to preserve this order. However, I can't find whether all ICU calendar implementations are part of the standard - where should I look for that?

Reply:
The ICU calendar implementations will be built-in to Intl-aware JavaScript engines, but they won't be _specified_, at least not as part of this proposal. That's because, like time zones, calendars may change for political and administrative reasons, and new calendars may be added to ICU.

I think the general consensus has been that built-in calendars should adhere to the same standards as the ISO calendar, but userland calendars are only "recommended" to do these things. So I do think that the implementations in the polyfill ought to do the same things as the ISO calendar, and I believe that's what #1245 does as well.

Reply:
Fixed in update

Reply:
Fixed in update